### PR TITLE
Feature/poetry

### LINKED ISSUE #156: Use poetry for dependency management
Whilst installing dependencies I have stumbled upon the following conflicts:

1. Python 3.6
   - current Python requirement (>=3.6,<4.0) is not compatible with `tortoise-orm`'s requirement of Python >=3.7,<4.0.
2.  `ormar` and `gino-starlette` have conflicting `importlib_metadata` version requirements.
3. `databases` and `gino-starlette` have conflicting requirements for `SQLAlchemy`.
4. `pytest` >= 7 has conflicting requirements.

My proposal — use [poetry](https://python-poetry.org/) for managing dependencies:

- it automatically resolves issues with dependencies;
- no need for creating virtual environment manually — poetry creates it for you;
- there are some other advantages.

Haven't seen any open/closed issues or pull requests on the subject, so opening this ticket.

### PR DESCRIPTION
**Pull request by dclimber**
Resolves #156 

- Adds `pyproject.toml` with poetry config.
- Adds `poetry.lock` — all dependencies are here.
- Updates `requirements.txt` and `dev.requirements.txt` — exported from poetry, so no conflicting package versions are installed.
- Updates `contributing.md` with info about poetry.
- Changes `psycopg2` to `psycopg2-binary` to avoid building `psycopg2`.

### CODE REVIEW

**Comment by dclimber** [comment]
Whoops, did not run the tests — my bad. There seems to be some issues with the resolved pydantic version. Working on it.

**Comment by awtkns** [comment]
Hi @dclimber, thanks for this! This is a solid improvement. My main concern here is dropping support for python 3.6. As `fastapi` supports python 3.6 so should this library. 

The one caveat here is that the [tortoise-orm](https://github.com/tortoise/tortoise-orm) CRUDRouter is not available to those using python 3.6.

Curious on your thoughts. Thanks again for thsi!

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,7 @@
-fastapi>=0.62.0
\ No newline at end of file
+anyio==3.6.1; python_version >= "3.6" and python_full_version >= "3.6.2"
```
Comment (awtkns) [inline requirements.txt:1]: Does this file need to be committed still with poetry?

Comment (dclimber) [reply requirements.txt:1]: > The one caveat here is that the [tortoise-orm](https://github.com/tortoise/tortoise-orm) CRUDRouter is not available to those using python 3.6.

Oh ok, get it now — so if one uses python 3.6, then they simply cannot use tortoise-orm, is that correct?
If so, I'll update the code to accommodate the feature.

> Does this file need to be committed still with poetry?

It is not necessary, but allows people, who don't use poetry to simply install dependencies with `pip`, so I've kept it.

Comment (awtkns) [reply requirements.txt:1]: > Oh ok, get it now — so if one uses python 3.6, then they simply cannot use tortoise-orm, is that correct?
If so, I'll update the code to accommodate the feature.

Yes this is correct!

**Comment by dclimber** [comment]
@awtkns phew everything works finally.

**Review by awtkns — APPROVED** [review]

**Comment by awtkns** [comment]
@dclimber I think the github workflows may need to be changed so that they support poetry

**Comment by dclimber** [comment]
@awtkns hmm yeah it seems. I have python 3.8 on my system, so things work smoothly, but other versions need testing.
I'll look into either workflow improvement with `poetry` or fixing things, so existing workflows pass.

**Comment by dclimber** [comment]
@awtkns pipelines finally pass! https://github.com/dclimber/fastapi-crudrouter/pull/3

**Comment by awtkns** [comment]
Thanks for the great work @dclimber. I've seen that you have been active on some other issues. Let me know if there is any way I can support 🚀

**Pull request merged**